src/chatd.cpp

```diff
@@ -5191,6 +5191,8 @@ void Chat::onUserJoin(Id userid, Priv priv)
         CALL_CRYPTO(onUserJoin, userid);
     }
 
+    ///TODO: in public link extra notification can be generated if we are re-invited
+    /// and chatd joins arrive before than api notification
```

Comment:
I think this is not the right place for the comment (that needs to indicate both, the auto-join from another client and the invitation by a moderator). The right place is where the notifications are filtered out, with some context about why they are skipped and why not in the cases aforementioned.

Reply:
Fixed